### ISSUE TITLE
Widget not showing display name

### ISSUE THREAD

**Issue by i701** [davidhi7/ddcci-plasmoid#3]
I followed the instructions and i don't see my external display name in the widget. However, i can adjust using manual commands. 

![image](https://user-images.githubusercontent.com/72292943/230905677-4839daf8-3e21-438e-91a6-aedf63ed8a20.png)

**Comment by mesonium** [comment]
Same issue. If you run `ddcutil detect` does it also say `DDC communication failed` for you?

**Comment by davidhi7** [comment]
Can you post the output of `python -m ddcci_plasmoid_backend --debug detect` and `ddccutil detect`?

**Comment by i701** [comment]
`➜  ~ python -m ddcci_plasmoid_backend --debug detect
DEBUG ddcci_plasmoid_backend: Run in debug mode
DEBUG __main__: Command: detect
DEBUG ddcci_plasmoid_backend.ddcci: Execute command: `ddcutil detect`
DEBUG __main__: Detected 0 working monitor busses, 1 non-working bus.
{"command": "detect", "value": []}
➜  ~ ddcutil detect
Unable to open directory /sys/bus/i2c/devices/i2c--1: No such file or directory
Device /dev/i2c-255 does not exist. Error = ENOENT(2): No such file or directory
/sys/bus/i2c buses without /dev/i2c-N devices: /sys/bus/i2c/devices/i2c-255
Driver i2c_dev must be loaded or builtin
See https://www.ddcutil.com/kernel_module
Invalid display
   I2C bus:  /dev/i2c-3
   DRM connector:           card0-eDP-1
   EDID synopsis:
      Mfg id:               BOE - BOE
      Model:                
      Product code:         1670  (0x0686)
      Serial number:        
      Binary serial number: 0 (0x00000000)
      Manufacture year:     2015,  Week: 1
   DDC communication failed
   This is an eDP laptop display. Laptop displays do not support DDC/CI.

Display 1
   I2C bus:  /dev/i2c-9
   DRM connector:           card1-HDMI-A-1
   EDID synopsis:
      Mfg id:               DEL - Dell Inc.
      Model:                DELL P2419H
      Product code:         53466  (0xd0da)
      Serial number:        4LT6G53
      Binary serial number: 843198796 (0x3242314c)
      Manufacture year:     2020,  Week: 38
   VCP version:         2.1

➜  ~ 
`

**Comment by davidhi7** [comment]
Your first 'invalid' display is a laptop display, setting their brightness is not currently supported with this widget since this requires different protocols. The 4. line in the output you posted indicates that the second display has been detected, however the DDC communication does not work. Specifically, `ddcci-plasmoid-backend` tries to fetch the current brightness from this display and if this fails, it is not included in the widget.

Can you try to set the brightness manually using ddcutil?
`ddcutil setvcp --display 1 0x10 100`
This should set the brightness to 100%.

**Comment by i701** [comment]
Yeah, manually it works like I mentioned before also. Just the widget does not work.

**Comment by i701** [comment]
```
➜  ~ python -m ddcci_plasmoid_backend --debug detect
DEBUG __main__: Running version 0.1.4 in debug mode
DEBUG __main__: Command: detect
DEBUG ddcci_plasmoid_backend.ddcci: Execute command: `ddcutil detect`
DEBUG ddcci_plasmoid_backend.ddcci: [code]   ddcutil detect: 0
DEBUG ddcci_plasmoid_backend.ddcci: [stdout] ddcutil detect: Unable to open directory /sys/bus/i2c/devices/i2c--1: No such file or directory
Device /dev/i2c-255 does not exist. Error = ENOENT(2): No such file or directory
/sys/bus/i2c buses without /dev/i2c-N devices: /sys/bus/i2c/devices/i2c-255
Driver i2c_dev must be loaded or builtin
See https://www.ddcutil.com/kernel_module
Invalid display
   I2C bus:  /dev/i2c-3
   DRM connector:           card0-eDP-1
   EDID synopsis:
      Mfg id:               BOE - BOE
      Model:                
      Product code:         1670  (0x0686)
      Serial number:        
      Binary serial number: 0 (0x00000000)
      Manufacture year:     2015,  Week: 1
   DDC communication failed
   This is an eDP laptop display. Laptop displays do not support DDC/CI.

Display 1
   I2C bus:  /dev/i2c-9
   DRM connector:           card1-HDMI-A-1
   EDID synopsis:
      Mfg id:               DEL - Dell Inc.
      Model:                DELL P2419H
      Product code:         53466  (0xd0da)
      Serial number:        4LT6G53
      Binary serial number: 843198796 (0x3242314c)
      Manufacture year:     2020,  Week: 38
   VCP version:         2.1
DEBUG ddcci_plasmoid_backend.ddcci: [stderr] ddcutil detect: 

DEBUG ddcci_plasmoid_backend.ddcci: Found 7 entries at root level
DEBUG ddcci_plasmoid_backend.ddcci: Key Unable to open directory /sys/bus/i2c/devices/i2c--1 does not match pattern for valid display, so skip it
DEBUG ddcci_plasmoid_backend.ddcci: Key Device /dev/i2c-255 does not exist. Error = ENOENT(2) does not match pattern for valid display, so skip it
DEBUG ddcci_plasmoid_backend.ddcci: Key /sys/bus/i2c buses without /dev/i2c-N devices does not match pattern for valid display, so skip it
DEBUG ddcci_plasmoid_backend.ddcci: Key Driver i2c_dev must be loaded or builtin does not match pattern for valid display, so skip it
DEBUG ddcci_plasmoid_backend.ddcci: Key See https does not match pattern for valid display, so skip it
DEBUG ddcci_plasmoid_backend.ddcci: Key Invalid display does not match pattern for valid display, so skip it
DEBUG ddcci_plasmoid_backend.ddcci: Execute command: `ddcutil getvcp --bus 9 --brief 10`
DEBUG ddcci_plasmoid_backend.ddcci: [code]   ddcutil getvcp --bus 9 --brief 10: 1
DEBUG ddcci_plasmoid_backend.ddcci: [stdout] ddcutil getvcp --bus 9 --brief 10: Unable to open directory /sys/bus/i2c/devices/i2c--1: No such file or directory
Device /dev/i2c-255 does not exist. Error = ENOENT(2): No such file or directory
/sys/bus/i2c buses without /dev/i2c-N devices: /sys/bus/i2c/devices/i2c-255
Driver i2c_dev must be loaded or builtin
See https://www.ddcutil.com/kernel_module
VCP 10 C 60 100
DEBUG ddcci_plasmoid_backend.ddcci: [stderr] ddcutil getvcp --bus 9 --brief 10: 

DEBUG __main__: too many values to unpack (expected 5)
DEBUG __main__: Detected 0 working monitor busses, 1 non-working bus.
{"command": "detect", "value": []}

```

**Comment by davidhi7** [comment]
Apparently ddcutil is printing some irrelevant error messages into stdout and returning 1, while actually doing its job of fetching the display brightness just fine (assuming your brightness was set to 60% when running the command). According to [this](https://github.com/rockowitz/ddcutil/issues/293) issue, this bug has been fixed in ddcutil v1.4.1. Can you check your ddcutil version and update it accordingly?

I could ignore all error messages and the error code when parsing the response. However, I think this is only going to cause other bugs in the future. Considering this is likely already fixed on the ddcutil side for months, I don't plan on implementing hacky bypasses into the backend.

**Comment by rburcham** [comment]
I can confirm: upgrading to ddcutil 1.4.1 solves this problem and the widget works as expected.
